### PR TITLE
default: Fix removing content from written books.

### PR DESCRIPTION
**Pull request by orbea**
Fixes https://github.com/minetest/minetest_game/issues/1743

This fixes saving a book without any text and / or title and removing the text and / or title from a written book when you are the owner.

I tested this in both singleplayer and multiplayer without issues.

* If a book does not have any text **and** title anyone can write to it even when they are not the owner and then assume ownership.
* If the book has any text **or** title then only the owner can write to it.
* As expected if both the text **and** title are removed then any player can add a new title and / or text.
* If the book does not have a title it will default to `"Book" by playername`.

This took inspiration from the old and closed PR https://github.com/minetest/minetest_game/pull/2157.

I have not considered adding a recipe to turn a written book into a blank book, if this is done I think its outside the scope of this PR and should be done in a new one.

### CODE REVIEW

File: mods/default/craftitems.lua [hunk 1]
```diff
@@ -87,16 +86,19 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 			data = stack:get_meta():to_table().fields
 		end
 
-		if data and data.owner and data.owner ~= player:get_player_name() then
+		if data and data.owner and data.owner ~= player:get_player_name()
+				and fields.title ~= "" and fields.text ~= "" then
```
Comment (SmallJoker) [inline mods/default/craftitems.lua:90]: Please separate these checks. This addition allows modifying the contents by players who aren't the owner of the book.

Comment (orbea) [reply mods/default/craftitems.lua:90]: @SmallJoker That was the intent, when a book no text **and** title anyone could take over ownership. If the book has any text **or** title then they will not be able to modify it. Do you disagree with this intent?

Comment (SmallJoker) [reply mods/default/craftitems.lua:90]: I thought it was only about erasing books where you have write permissions (i.e. are the owner of the book). Thus, my conclusions that this change introduced an unwanted behaviour.
In this case it's fine.

**Comment by orbea** [comment]
@SmallJoker I'm happy to change that design goal when books have both empty text and title if you think that is better. My mail goal is to make it possible to save book that have both empty text and / or title and to make it possible to remove the text or title from a written book (By the author).

**Comment by orbea** [comment]
Replaced by https://github.com/minetest/minetest_game/pull/2656.

File: mods/default/craftitems.lua (hunk 1 above)
Comment (paramat) [reply mods/default/craftitems.lua:90]: > This addition allows modifying the contents by players who aren't the owner of the book.

Seems wrong to me, even if the book has no text or title.
If a book with no text or title is reverted to default:book, which i prefer, then this is not needed.

**Pull request closed without merging**